src/Responses/HooksListResponse.php

```diff
@@ -27,23 +27,20 @@
  */
 class HooksListResponse extends BaseResponse
 {
-    /**
-     * @var Hook[]
-     */
-    private $hooks;
 
     /**
      * @return Hook[]
      */
-    public function getHooks()
+    public function getHooks(): array
     {
-        if (null === $this->hooks) {
-            $this->hooks = [];
-            foreach ($this->rawXml->hooks->children() as $hookXml) {
-                $this->hooks[] = new Hook($hookXml);
+        $hooks = [];
```

Comment:
Trade-off between more memory (old version) and more cpu (new version). Need a good argument to decided which solution to choose for the community. We assume the XML content is static and never changes in a single API call.

Reply:
Technically you are right (memory vs. cpu), but I guess those API-small data do no harm to any of them in these days. This proposal is just a cleaner and leaner way ... and it is more robust against programming errors (less code, less variables, low risk of inconsistent data).